### ISSUE TITLE
CTD on 1.4 beta branch

### ISSUE THREAD

**Issue by chrisjon** [Tyler-IN/MnB2-Bannerlord-CommunityPatch#286]
as of 11-5-2020 on beta branch, clicking the clan icon crashes the game to desktop, worked before 1.4 beta ninja patch on 11-5-2020

Edit. to replicate, just create a fresh campaign on e1.4.0.228869 skip the tutorials, and click the clan icon.

**Comment by iPherian** [comment]
I was lucky enough that my steam hadn't updated yet.

With CP loaded:
Pre ninja patch: no crash
post ninja patch: crashes

If no mods enabled no crash either way.

When debugger attached in a debug build, exception trace:

```
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.MissingMethodException: Method not found: 'System.Collections.Generic.IReadOnlyList`1<TaleWorlds.CampaignSystem.Town> TaleWorlds.CampaignSystem.Town.get_All()'.
  at at CommunityPatch.Patches.Perks.Intelligence.Steward.RulerPatch.CompanionLimitPatched(Clan __instance, Int32& __result)
  at at DMD<DMD<get_CompanionLimit_Patch2>?16294837::get_CompanionLimit_Patch2>(Clan this)
  at at TaleWorlds.CampaignSystem.ViewModelCollection.ClanManagement.Categories.ClanMembersVM.RefreshMembersList()
  at at TaleWorlds.CampaignSystem.ViewModelCollection.ClanManagement.Categories.ClanMembersVM..ctor(Action onRefresh)
  at at TaleWorlds.CampaignSystem.ViewModelCollection.ClanManagement.ClanVM..ctor(Action onClose, Action`1 openPartyAsManage, Action openBannerEditor)
  at at SandBox.GauntletUI.GauntletClanScreen.TaleWorlds.Core.IGameStateListener.OnActivate()
  at at TaleWorlds.Core.GameState.HandleActivate()
  at at TaleWorlds.Core.GameStateManager.OnPushState(GameState gameState)
  at at TaleWorlds.Core.GameStateManager.DoGameStateJobs()
  at at TaleWorlds.Core.GameStateManager.PushState(GameState gameState, Int32 level)
  at at SandBox.View.Map.MapNavigationHandler.OpenClanScreenAction()
  at at SandBox.View.Map.MapNavigationHandler.TaleWorlds.CampaignSystem.INavigationHandler.OpenClan()
  at at TaleWorlds.CampaignSystem.ViewModelCollection.Map.MapNavigationVM.ExecuteOpenClan()
  --- End of inner exception stack trace ---
  at at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
  at at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
  at at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
  at at TaleWorlds.Library.Common.InvokeWithLog(MethodInfo methodInfo, Object obj, Object[] args)
  at at TaleWorlds.GauntletUI.Data.GauntletView.OnCommand(String command, Object[] args)
  at at System.Action`3.Invoke(T1 arg1, T2 arg2, T3 arg3)
  at at TaleWorlds.GauntletUI.Widget.EventFired(String eventName, Object[] args)
  at at TaleWorlds.GauntletUI.ButtonWidget.HandleClick()
  at at TaleWorlds.GauntletUI.ButtonWidget.OnMouseReleased()
  at at TaleWorlds.GauntletUI.EventManager.MouseUp()
  at at TaleWorlds.GauntletUI.UIContext.UpdateInput(InputType handleInputs)
  at at TaleWorlds.Engine.GauntletUI.GauntletLayer.ProcessEvents()
  at at TaleWorlds.Engine.Screens.ScreenManager.Update()
  at at TaleWorlds.Engine.Screens.ScreenManager.Tick(Single dt)
  at at ManagedCallbacks.EngineCallbacksGenerated.ScreenManager_Tick(Single dt)
```

**Comment by iPherian** [comment]
As the stacktrace says, it does indeed appear that the getter for property TaleWorlds.CampaignSystem.Town.All has now been removed. Rider gives a compilation error.

I love TW but I think they just removed a public method in a stealth ninja patch 🤣 Oh the humanity

**Comment by Solganis** [comment]
Confirmed. After ninja 1.4 Beta patch today CTD when opening a "Clan" screen.

```
Source: CommunityPatch 
Method not found: 'System.Collections.Generic.IReadOnlyList`1 TaleWorlds.CampaignSystem.Town.get_All()'. 

at CommunityPatch.Patches.Perks.Intelligence.Steward.RulerPatch.CompanionLimitPatched(Clan __instance, Int32& __result)
   at DMD?10603613::get_CompanionLimit_Patch2>(Clan this)
   at TaleWorlds.CampaignSystem.ViewModelCollection.ClanManagement.Categories.ClanMembersVM.RefreshMembersList()
   at TaleWorlds.CampaignSystem.ViewModelCollection.ClanManagement.Categories.ClanMembersVM..ctor(Action onRefresh)
   at TaleWorlds.CampaignSystem.ViewModelCollection.ClanManagement.ClanVM..ctor(Action onClose, Action`1 openPartyAsManage, Action openBannerEditor)
   at SandBox.GauntletUI.GauntletClanScreen.TaleWorlds.Core.IGameStateListener.OnActivate()
   at TaleWorlds.Core.GameState.HandleActivate()
   at TaleWorlds.Core.GameStateManager.OnPushState(GameState gameState)
   at TaleWorlds.Core.GameStateManager.DoGameStateJobs()
   at SandBox.View.Map.MapScreen.OpenClanScreen()
   at SandBox.View.Map.MapScreen.TaleWorlds.CampaignSystem.IMapStateHandler.AfterWaitTick(Single dt)
   at TaleWorlds.CampaignSystem.MapState.OnMapModeTick(Single dt)
   at TaleWorlds.Core.GameStateManager.OnTick(Single dt)
   at TaleWorlds.Core.Game.OnTick(Single dt)
   at TaleWorlds.Core.GameManagerBase.OnTick(Single dt)
   at DMD?66083951::OnApplicationTick_Patch1>(Module this, Single dt)

```

**Comment by chrisjon** [comment]
> As the stacktrace says, it does indeed appear that the getter for property `TaleWorlds.CampaignSystem.Town.All` has now been removed. Rider gives a compilation error.
> 
> I love TW but I think they just removed a public method in a stealth ninja patch 🤣 Oh the humanity

i just love that after several weeks, an update for community patch comes out, and seconds later TW destroys it with 1 quick ninja patch :)

**Comment by Solganis** [comment]
> 
> 
> > As the stacktrace says, it does indeed appear that the getter for property `TaleWorlds.CampaignSystem.Town.All` has now been removed. Rider gives a compilation error.
> > I love TW but I think they just removed a public method in a stealth ninja patch 🤣 Oh the humanity
> 
> i just love that after several weeks, an update for community patch comes out, and seconds later TW destroys it with 1 quick ninja patch :)

Exactly. It's a pretty bad time for a rc5 release on Nexus.
Who woulda thunk? Frickin hell.

**Comment by TYoung86** [comment]
Town.All is also checked in the Ruler patch.

**Comment by iPherian** [comment]
It looks like this code first showed up in commit a24e0dc8942c9c32ca0e296d7890f70a13de5bd1 which first appeared in version 0.0.2 a long time ago.

**Comment by PackOfDingoes** [comment]
Hey I was wondering if you implemented the fix anywhere yet? I know it has been literally only 30 minutes, but I have no idea how to build your github version or anything, feel free to tell me to just wait :P

**Comment by iPherian** [comment]
@PackOfDingoes we had some people around to fix it immediately ;)

the fix is on dev branch, you can build it using the instructions on the main page.

**Comment by iPherian** [comment]
@PackOfDingoes the fix is in [0.1.0-rc4](https://github.com/Tyler-IN/MnB2-Bannerlord-CommunityPatch/releases/tag/v0.1.0-rc4), you can install manually in vortex

**Comment by PackOfDingoes** [comment]
tyvm, I was thinking "oh no, time to actually follow instructions instead of just copy/pasting", but you've come to my rescue lol.
